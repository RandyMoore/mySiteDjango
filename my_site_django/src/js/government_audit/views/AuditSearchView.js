import React from 'react';
import ReactPaginate from 'react-paginate';

function Options(props) {
  return (
    <div>
      <a href="https://www.postgresql.org/docs/9.6/static/textsearch-controls.html"> Query Parser: </a>
      <select value={props.queryParser} onChange={props.onParserChange}>
        <option value="plain">plain</option>
        <option value="phrase">phrase</option>
        <option value="raw">tsquery raw</option>
      </select>
    </div>
  );
}

function SearchInput(props) {
  return (
    <form onSubmit={props.onQuerySubmit}>
      <label>
        Search Query <br/>
        <input type="text" value={props.searchResults.query} onChange={props.onQueryChange}/>
      </label>
      <input type="submit" value="Submit" />
    </form>
  );
}

function ResultsTable(props) {
  return (
    <table className="results-table">
      <thead>
        <tr>
          <th/>
          <th width="80%">Title</th>
          <th>Publication Date</th>
         </tr>
       </thead>
       <tbody>
         {props.searchResults.results && props.searchResults.results.map(function(result, index) {
           return <tr key={ index }>
             <td>{ props.searchResults.resultsOffset + index + 1 }</td>
             <td>
               <a href={ result.get('url') }>{ result.get('title') }</a>
             </td>
             <td> {result.get('date')} </td>
           </tr>;
         })}
       </tbody>
    </table>
  );
}

function AuditSearch(props) {
  const pageCount = Math.ceil(props.searchResults.resultsSize / props.searchResults.resultsLimit)
  return (
    <div>
      <SearchInput {...props} />
      <Options {...props} />
      <ResultsTable {...props} />
      { pageCount > 1 &&
        <ReactPaginate {...props}
          previousLabel={"previous"}
          nextLabel={"next"}
          breakLabel={<a>...</a>}
          breakClassName={"break-me"}
          pageCount={pageCount}
          marginPagesDisplayed={2}
          pageRangeDisplayed={5}
          onPageChange={props.onPageChange}
          containerClassName={"pagination"}
          subContainerClassName={"pages pagination"}
          activeClassName={"active"}
        />
      }
    </div>
  );
}

export default AuditSearch;
