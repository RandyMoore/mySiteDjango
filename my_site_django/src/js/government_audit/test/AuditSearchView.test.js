import React from 'react';
import renderer from 'react-test-renderer';
import AuditSearchView from '../views/AuditSearchView.js';

test('Default render', () => {
  const searchResults = {
    'resultsSize': 0,
    'resultsLimit': 10
  }

  const component = renderer.create(
    <AuditSearchView
      searchResults={searchResults}
    />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render with pagination visible', () => {
  const searchResults = {
    'resultsSize': 11,
    'resultsLimit': 10
  }

  const component = renderer.create(
    <AuditSearchView
      searchResults={searchResults}
    />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
