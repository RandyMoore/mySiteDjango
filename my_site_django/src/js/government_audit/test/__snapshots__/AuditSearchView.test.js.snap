// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render 1`] = `
<div>
  <form
    onSubmit={undefined}
  >
    <label>
      Search Query 
      <br />
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <div>
    <a
      href="https://www.postgresql.org/docs/9.6/static/textsearch-controls.html"
    >
       Query Parser: 
    </a>
    <select
      onChange={undefined}
      value={undefined}
    >
      <option
        value="plain"
      >
        plain
      </option>
      <option
        value="phrase"
      >
        phrase
      </option>
      <option
        value="raw"
      >
        tsquery raw
      </option>
    </select>
  </div>
  <table
    className="results-table"
  >
    <thead>
      <tr>
        <th />
        <th
          width="80%"
        >
          Title
        </th>
        <th>
          Publication Date
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Render with pagination visible 1`] = `
<div>
  <form
    onSubmit={undefined}
  >
    <label>
      Search Query 
      <br />
      <input
        onChange={undefined}
        type="text"
        value={undefined}
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <div>
    <a
      href="https://www.postgresql.org/docs/9.6/static/textsearch-controls.html"
    >
       Query Parser: 
    </a>
    <select
      onChange={undefined}
      value={undefined}
    >
      <option
        value="plain"
      >
        plain
      </option>
      <option
        value="phrase"
      >
        phrase
      </option>
      <option
        value="raw"
      >
        tsquery raw
      </option>
    </select>
  </div>
  <table
    className="results-table"
  >
    <thead>
      <tr>
        <th />
        <th
          width="80%"
        >
          Title
        </th>
        <th>
          Publication Date
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
  <ul
    className="pagination"
  >
    <li
      className="previous disabled"
    >
      <a
        className={undefined}
        href={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
      >
        previous
      </a>
    </li>
    <li
      className="active"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        className={undefined}
        href={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
      >
        1
      </a>
    </li>
    <li
      className={undefined}
    >
      <a
        aria-current={null}
        aria-label="Page 2"
        className={undefined}
        href={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
      >
        2
      </a>
    </li>
    <li
      className="next"
    >
      <a
        className={undefined}
        href={undefined}
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
      >
        next
      </a>
    </li>
  </ul>
</div>
`;
